### PR TITLE
fix: .env challenge and related

### LINKED ISSUE #40915: Use the .env File Challenge Issue
For more coverage, I am opening this here:

Challenge: https://www.freecodecamp.org/learn/apis-and-microservices/basic-node-and-express/use-the--env-file

As of <04/02/2021, a change has caused the above mentioned lesson to be uncompletable.

Error: `SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at /home/runner/boilerplate-express-2/node_modules/fcc-express-bground/index.js:91:26`

The error in full:
<details>
<summary>Error</summary>

```
SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at /home/runner/boilerplate-express-2/node_modules/fcc-express-bground/index.js:91:26
    at IncomingMessage.<anonymous> (/home/runner/boilerplate-express-2/node_modules/fcc-express-bground/index.js:33:7)
    at IncomingMessage.emit (events.js:315:20)
    at IncomingMessage.Readable.read (_stream_readable.js:505:10)
    at flow (_stream_readable.js:1005:34)
    at resume_ (_stream_readable.js:986:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at /home/runner/boilerplate-express-2/node_modules/fcc-express-bground/index.js:99:28
    at IncomingMessage.<anonymous> (/home/runner/boilerplate-express-2/node_modules/fcc-express-bground/index.js:33:7)
    at IncomingMessage.emit (events.js:315:20)
    at IncomingMessage.Readable.read (_stream_readable.js:505:10)
    at flow (_stream_readable.js:1005:34)
    at resume_ (_stream_readable.js:986:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
_http_outgoing.js:518
    throw new ERR_HTTP_HEADERS_SENT('set');
    ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at ServerResponse.setHeader (_http_outgoing.js:518:11)
    at ServerResponse.header (/home/runner/boilerplate-express-2/node_modules/express/lib/response.js:771:10)
    at ServerResponse.send (/home/runner/boilerplate-express-2/node_modules/express/lib/response.js:170:12)
    at ServerResponse.json (/home/runner/boilerplate-express-2/node_modules/express/lib/response.js:267:15)
    at /home/runner/boilerplate-express-2/node_modules/fcc-express-bground/index.js:108:15
    at IncomingMessage.<anonymous> (/home/runner/boilerplate-express-2/node_modules/fcc-express-bground/index.js:33:7)
    at IncomingMessage.emit (events.js:315:20)
    at IncomingMessage.Readable.read (_stream_readable.js:505:10)
    at flow (_stream_readable.js:1005:34)
    at resume_ (_stream_readable.js:986:3) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
```

</details>

If you encounter this, the best course of action is to skip the lesson, and continue with the curriculum.

As nothing obvious has changed on freeCodeCamp's side, this is likely an issue on Repl.it's side.

<details>
<summary>Current Debugged Steps</summary>

1) Tested code which previously used to pass - no longer passes
2) Downgraded `express` package - no effect
3) One user was able to pass, and the noticeable difference was their `origin` was not `freecodecamp.org`, when they submitted.
4) All manor of sending data via stringifying etc.

</details>

From forum post: https://forum.freecodecamp.org/t/headers-error-in-env-challenge/444357

__P.S. Even though I have said there has been no change to the challenge, in the last few seconds (19:00 UTC) there has been a change pushed to `main`. Otherwise, our side has not changed in a way I can see which would affect this_

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/40915

Followed much of the advice given by @gikf , and managed to complete the challenges which posed issue.

Here is the project I used for testing:
https://repl.it/join/xcpucqrx-sky020

### CODE REVIEW

**Comment by lasjorg** [comment]
It does seem to work, but it will also prevent you from working locally using HTTP. I get the same crash locally with your code as I do on repl.it when first testing using `app.set('trust proxy', true)`.

So I looked and oddly enough without the use of `selfCaller` it works just fine, i.e. the `/_api/hello-console` route works with `app.set('trust proxy', true)`. So looking more at the code it seems we just need to set the port correctly for it to work.


TL;DR

If we set the port correctly and use `app.set('trust proxy', true)` it seems to work for me (with no other change to the code).

```
var opts = {
  hostname : url,
  method : 'GET',
  path : path,
  port : req.protocol === 'https' ? 443 : port || 80,
  headers: {
    'User-Agent':
      'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.106 Safari/537.36',
  },
};
```

```
function setupBackgroundApp(app, myApp, dirname) {
  app.set('trust proxy', true);
  
...rest of code
}
```

---

I tested all the challenges and they all passed except the "Serve static assets" for some reason (I get a 404). I didn't really want to start looking at what was wrong but the CSS does load on the page.
https://repl.it/@lasjorg/Basic-Node-and-Express-LIB-test-port-and-appset
